.gitlab-ci.yml

```diff
@@ -111,31 +111,34 @@ hackage-release:
   rules:
     - if: '$CI_COMMIT_TAG != null' # tags
 
-# Work around https://gitlab.com/gitlab-org/gitlab/-/issues/216629
+# We manually report status to GitHub to work around
+# <https://gitlab.com/gitlab-org/gitlab/-/issues/216629>.
+# The script will also send a Slack message when the nightly fails.
 #
 # If we ever remove this, we may have to rethink the use of the interruptible flag
```

Comment:
If we ever remove /this/. What's /this/? The jobs that manually report the status to GitHub? Or something more specific?

Reply:
Yes, _this_ refers to manually reporting status to GitHub.

From 70b7b2ecb4647163d807a30072cc1faf922a3710:
> [..]
> In the past we've disabled this(tasks being aborted by new commits being pushed) because aborted tasks looked like failed tasks on GitHub.

With automatic status reporting and 
With automatic task interruption, if you push more commits to a PR before the CI finished testing, it will abort the CI jobs running on the previous HEAD commit of the branch and restart the CI jobs on the new HEAD of the branch.
But with automatic status reporting these aborted jobs are shown as :x: on the PR page.

It can look like this:
* first commit `xxxxxxxx`
* second commit :x: `yyyyyyyy`
* third commit :heavy_check_mark: `zzzzzzzzzz`
 
Suggesting that the second commit failed CI, while in fact it just didn't finish before it got interrupted.